### PR TITLE
Pipeline i18n refactor

### PR DESCRIPTION
**Pull request by andrewballantyne**
https://issues.redhat.com/browse/ODC-5778

Fixes the excessive passing of `t` everywhere inside the Pipeline project.

Two major takeaways:

1. I didn't touch anything outside of the pipelines-plugin package, thus there is the odd usage of passing t (although now it is `i18n.t`) still in Pipelines but that goes off to the shared package (which is used by many packages and thus seemed unnecessary at this juncture)
    * Searched for all `TFunction` type usages and removed them in all cases that were used in function arguments
2. I added global mocks for `i18next` and `react-i18next` to reduce boilerplate mocking in all of our tests -- more on that can be found by reading [manual-mocks from jest](https://jestjs.io/docs/manual-mocks)
    * Others should be able to just remove the mocks in their test files as they are duping the mock effort -- shouldn't hurt that they are duped for the time being
    * Mock differs slightly from normal mocks as it splits on `~` to provide a more realistic usage; `labelKey` usages in Pipelines were unaffected by this as they circumvent the `t` invocations for any tests that use them

### CODE REVIEW

File: frontend/packages/pipelines-plugin/src/components/pipelines/pipeline-builder/task-sidebar/TaskSidebarName.tsx [hunk 1]
```diff
@@ -46,7 +47,7 @@ const TaskSidebarName: React.FC<TaskSidebarNameProps> = (props) => {
           }
 
           setValidating(true);
-          nameValidationSchema(t, 63)
+          nameValidationSchema((tKey) => i18n.t(tKey), 63)
```
Comment (andrewballantyne) [inline frontend/packages/pipelines-plugin/src/components/pipelines/pipeline-builder/task-sidebar/TaskSidebarName.tsx:50]: Eg of a half-way change. If this is undesirable, I can make the change to the shared folder and in the 4-5 other packages it impacts.

File: frontend/packages/pipelines-plugin/src/components/pipelines/pipeline-overview/PipelineOverview.spec.tsx [hunk 2]
```diff
@@ -29,7 +29,7 @@ describe('Pipeline sidebar overview', () => {
 
   it('should show view all link if there are more than MAX_VISIBLE pipelineruns', () => {
     props.item.pipelineRuns = ['pr0', 'pr1', 'pr2', 'pr3'].map((pr) => ({
-      metadata: { name: pr, namespace: 'test' },
+      metadata: { name: pr, namespace: 'test', uid: pr },
```
Comment (andrewballantyne) [inline frontend/packages/pipelines-plugin/src/components/pipelines/pipeline-overview/PipelineOverview.spec.tsx:32]: React was echo'ing key in list errors... striving to make it a clean log view.

File: frontend/packages/pipelines-plugin/src/components/pipelines/detail-page-tabs/pipeline-details/TaskComponentTaskStatus.tsx [hunk 3]
```diff
@@ -1,26 +0,0 @@
...
-      color: getRunStatusColor(runStatus, t).pftoken.value,
-      name,
-      size: 1,
-    };
-  });
-
-  return <HorizontalStackedBars values={visualValues} barGap={2} height={2} />;
-};
-
-export default TaskComponentTaskStatus;
```
Comment (andrewballantyne) [inline frontend/packages/pipelines-plugin/src/components/pipelines/detail-page-tabs/pipeline-details/TaskComponentTaskStatus.tsx:26]: Unused component -- removed it when I was adjusting i18n

File: frontend/packages/pipelines-plugin/src/components/pipelines/pipeline-overview/TriggerLastRunButton.tsx [hunk 4]
```diff
@@ -44,7 +46,7 @@ const TriggerLastRunButton: React.FC<TriggerLastRunButtonProps> = ({
         onClick={callback}
         isDisabled={pipelineRuns.length === 0 && !callback}
       >
-        {labelKey}
+        {t(labelKey)}
```
Comment (andrewballantyne) [inline frontend/packages/pipelines-plugin/src/components/pipelines/pipeline-overview/TriggerLastRunButton.tsx:49]: This is a legit change - for any reviewers.

File: frontend/packages/pipelines-plugin/src/utils/pipeline-actions.tsx [hunk 5]
```diff
@@ -119,11 +119,12 @@ const rerunPipeline: KebabAction = (
...
   resources: any,
-  customData: RerunPipelineData = { labelKey: 'Start Last Run' },
+  customData: RerunPipelineData = {},
 ) => {
-  const { onComplete } = customData;
+  // t('pipelines-plugin~Start last run')
+  const { labelKey = 'pipelines-plugin~Start last run', onComplete } = customData;
 
-  const sharedProps = { labelKey: customData.labelKey, accessReview: {} };
+  const sharedProps = { labelKey, accessReview: {} };
```
Comment (andrewballantyne) [inline frontend/packages/pipelines-plugin/src/utils/pipeline-actions.tsx:127]: This is the functionality of that change. We didn't translate the labelKey when it was default.

**Review by andrewballantyne — COMMENTED** [review]
Because of the large PR... just wanted to note an actual change for ease of reviewing.

**Comment by karthikjeeyar** [comment]
Verified the changes around pipeline code refactoring, looks good to me.

**Comment by christianvogt** [comment]
global mocks lgtm

**Pull request merged**